protocol/communities_messenger_test.go

```diff
@@ -108,9 +110,9 @@ func (s *MessengerCommunitiesSuite) SetupTest() {
 	s.shh = gethbridge.NewGethWakuWrapper(shh)
 	s.Require().NoError(shh.Start())
 
-	s.admin = s.newMessenger(AdminPassword)
-	s.bob = s.newMessenger(BobPassword)
-	s.alice = s.newMessenger(AlicePassword)
+	s.admin = s.newMessenger(adminPassword, adminAddress)
+	s.bob = s.newMessenger(bobPassword, bobAddress)
+	s.alice = s.newMessenger(alicePassword, aliceAddress)
```

Comment:
The password and address is used to create a `AccountManagerMock`, the testable element is the `account.Manager` not the password or the address. So the customisable element should be on the `account.Manager` level not the password and address.

This is important as the current approach limits the reusability of `s.newMessenger` as it requires a password and address when they are not always needed.

```go
func makeAccountMockManager(password, walletAddress string){
	accountsManager := &AccountManagerMock{}
	accountsManager.AccountsMap = make(map[string]string)
	accountsManager.AccountsMap[walletAddress] = types.EncodeHex(crypto.Keccak256([]byte(password)))

}
```

```go
	s.admin = s.newMessenger()
	s.admin.accountsManager = makeAccountMockManager(adminPassword, adminAddress)
	s.bob = s.newMessenger(bobPassword, bobAddress)
	s.bob.accountsManager = makeAccountMockManager(bobPassword, bobAddress)
	s.alice = s.newMessenger(alicePassword, aliceAddress)
	s.alice.accountsManager = makeAccountMockManager(alicePassword, aliceAddress)
```

Reply:
Thanks, thats a good point! I've added additional `newMessengerWithWallet` function, because `accountsManager` needed by `communityManager`,  which is inside of a `Messenger`